src/silly_sprites/event_system.hpp

```diff
@@ -0,0 +1,67 @@
+#include "events.hpp"
+#include "types.hpp"
+#include <vector>
+
+namespace sly::event {
+
+    struct EventID final {
+        friend class EventSystem;
+
+    private:
+        usize m_id;
+        EventID(usize id) : m_id{ id } {};
+
+    public:
+        EventID() : m_id{ 0 } {};
```

Comment:
```suggestion
        constexpr EventID() : m_id{ 0 } {}
```
But do we need this constructor at all?

Reply:
i added this so that other object that hold a event id can be default constucted with an invalid id.
does this makes sense?

Reply:
From a technical standpoint it makes sense, yes. But should objects hold an invalid ID?

Reply:
we should discuss that.
when i remove this ctor you always have to write an ctor for every other object.
and you would have to call the `add_handler()` function in the initializerlist because the object has to be constructed before the body.
would wou be fine with that?
sometimes the initializerlist gets verry messy that way, when you have to define all llambdas in it. it would look cleaner when one would call the `add_handler()` function in the body.
on the other hand there would be no invalid ids at all when i remove this ctor.
what do you think?

Reply:
Oh, I think I didn't understand the problem before! But it's about objects that want to add a handler and later remove the handler. They have to store the ID somewhere. Am I right?
Okay, then I guess we will need some default constructor, because -- as you said -- otherwise working with this data type is really cumbersome.
What do you think of representing invalid IDs as an empty optional? The approach of "zero is invalid" seems a bit "un-modern".

Reply:
how does this work with the spaceship operator?
would this work?
if not i could define a `operator==` and let it handle the optionals.

when i return an empty optional if the id is 0, i have to return an optional with value when the id is valid right?

Reply:
I just saw that `tl::optional` should automatically support `operator<=>`. So this should be possible:
```cpp
[[nodiscard]] constexpr auto operator<=>(EventID const& other) const {
  return *this <=> other;
}
```